app/assets/javascripts/listings/components/listingContainer.js.coffee

```diff
@@ -25,14 +25,21 @@ angular.module('dahlia.components')
       @lotteryResultsListings = ListingService.lotteryResultsListings
 
       @isOwnershipListing = (listing) ->
-        @listing.Tenure == 'New sale' || @listing.Tenure == 'Resale'
+        listing.Tenure == 'New sale' || listing.Tenure == 'Resale'
 
-      @isOpenMatchListing = (listing) ->
-        @openMatchListings.indexOf(listing) > -1
+      @hasOwnAndRentFavorited = (listings) ->
+        ownershipFavorites = (@isOwnershipListing listing for listing in @filterByFavorites listings)
+        (_.some ownershipFavorites) && !(_.every ownershipFavorites)
```

Comment:
Having trouble understanding what the `some` and `every` means here. Does this mean if there are ownershipFavorites (`_.some ownershipFavorites`), and if not all of the `ownershipFavorites` are ownership listings then they must be rent favorites?

Wonder if we can include some clarification as a comment, or re-write to be clearer.

Reply:
I think @akegan wanted to optimize here. So intended functionality is achieved in 3 array passes instead of 4 in straight forward approach.

Reply:
True, I wanted to optimize. I could add a comment if that's helpful!

Reply:
@chrisdolendo let me know if you have any better ideas for a variable name here too.

it's a bool array of trues for ownership listings and falses for rental listings, it's also filtered by favorites in this line

Reply:
@chrisdolendo what about "areOwnershipFavorites"?

```
areOwnershipFavorites = (@isOwnershipListing listing for listing in @filterByFavorites listings)
# User has ownership and rental listings favorited if some but not all listings are ownership.
(_.some areOwnershipFavorites) && !(_.every areOwnershipFavorites)
```

Reply:
here's another option 
```
      @hasOwnAndRentFavorited = (listings) ->
        favoritedListings = @filterByFavorites listings
        areOwnershipListings = (@isOwnershipListing listing for listing in favoritedListings)
        # User has ownership and rental listings favorited if some but not all favorited listings are ownership.
        (_.some areOwnershipListings) && !(_.every areOwnershipListings)
```